client-v2/src/shared/components/input/InputImage.tsx

```diff
@@ -289,31 +293,31 @@ class InputImage extends React.Component<ComponentProps, ComponentState> {
 
   private handlePasteImgSrcChange = (e: React.FormEvent<HTMLInputElement>) => {
     e.preventDefault();
-    this.setState({ imageSrc: e.currentTarget.value });
+    e.persist();
```

Comment:
Check what this does. Do we need it here?

Reply:
`persist()` ensures that the event is available outside of its immediate context. It's a bit of an implementation detail from React -- React uses its own synthetic event handlers under the hood, and re-uses event objects liberally, so unless you call `persist()` when passing events elsewhere you can end up in situations where your event has disappeared. That probably sounds like evil magic. It kind of is. See https://reactjs.org/docs/events.html

I can't see that this event is being passed anywhere outside of this function, so you're probably safe to remove. You'll see red ink in the console if there are problems, so I'd remove, test and if all looks good then you're 👍

Reply:
Just as a pointless NB, other vdom libraries like Preact don't bother with this synthetic event stuff, which I think was largely to smooth out event handling irregularities between browsers. So it doesn't have to be this way! It just _happens_ to be this way for React.

Reply:
cool 👍 thanks for the explanation